Dashboard: Section add, remove, move up, and move down

psealock opened the pull request:
Fixes https://github.com/woocommerce/woocommerce-admin/issues/2122
Fixes https://github.com/woocommerce/woocommerce-admin/issues/2121

Add the ability to add/remove Dashboard Sections as well as move them up/down.

### Testing

1. Go to Dashboard.
2. In a Section's Ellipsis Menu, remove a section or move it up/down
3. Add a section back by going to the "plus" button at the bottom
<img width="425" alt="Screen Shot 2019-05-02 at 8 33 45 PM" src="https://user-images.githubusercontent.com/1922453/57064168-a1f04980-6d19-11e9-90ab-6176e645c8d6.png">
4. Play around and see if anything breaks.

psealock committed:
Dashboard: add section toggle

psealock commented on packages/components/src/ellipsis-menu/index.js:
```diff
@@ -33,8 +33,12 @@ class EllipsisMenu extends Component {
 
-		const renderContent = () => (
-			<NavigableMenu className="woocommerce-ellipsis-menu__content">{ children }</NavigableMenu>
+		// @todo Make all children rendered by render props so Dropdown args can be passed?
```
`EllipsisMenu` is essentially a wrapper for `Dropdown`, but by rendering children instead of using render props [like `Dropdown` does](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/dropdown/index.js#L94), we're removing built in functionality. In this case, use of argument `onToggle` so that dropdown contents can have access to this argument.

In a follow up PR, I'd like to change `EllipsisMenu` such that it accepts a `renderContent` prop instead of children.

Aljullu commented on client/dashboard/components/section-controls/style.scss:
```diff
@@ -0,0 +1,14 @@
+		vertical-align: bottom;
+	}
+
+	.woocommerce-ellipsis-menu__item div {
```
What would you think about rewriting this CSS ruleset, so we don't need the wrapping `<div>`?
```CSS
.woocommerce-ellipsis-menu__item {
	padding-bottom: 12px;
}
```

Aljullu commented on client/dashboard/dashboard-charts/index.js:
```diff
@@ -76,21 +77,27 @@ class DashboardCharts extends Component {
+								</MenuItem>
+							);
+						} ) }
+						<SectionControls onToggle={ onToggle } { ...this.props } />
```
I'm a bit afraid passing down all props (`{ ...this.props }`) to the `SectionControls` will become messy in the future as we keep adding new props to each section. How about declaring them one-by-one? It makes code longer and more verbose but I think it will help to understand which props are required by `SectionControls` and which don't.

Aljullu reviewed (commented):
Wow, this looks amazing and works like a charm! :heart_eyes: 

One suggestion: could we move sections to the bottom when they are enabled? If I have sections in this order:
1. _Performance_
2. _Charts_
3. _Leaderboards_

Imagine I remove _Performance_ and add it again. I would expect _Performance_ to be added below the _Leaderboards_, but currently it keeps the position it had before being removed. Not sure if it's only me but that felt a bit counter-intuitive to me.

In case a gif makes it clearer:
![order](https://user-images.githubusercontent.com/3616980/57067483-73e61780-6ccf-11e9-8b8f-8ede5e0ec337.gif)

psealock commented:
@Aljullu Thanks for the review.

I changed the CSS rule to remove an extra `div`, removed `...this.props` in favor of being verbose, and changed the behavior such that added Sections get appended to the bottom of the dashboard.

I also rebased with master.

Aljullu reviewed (approved):
Thanks for the changes @psealock! Looking good. :shipit:
